cmd/idracsdkapps/elasticsearch/elasticsearchplugin.go

```diff
@@ -0,0 +1,336 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/elastic/go-elasticsearch/v8/esapi"
+	"github.com/elastic/go-elasticsearch/v8"
+)
+
+var mtx sync.Mutex
+var dat map[string]interface{}
+var reportCount = 0
+var invalidJsonPkts = 0
+
+var telemetryserverIP = "100.98.72.157"
+
+type stagDellData struct {
+	ServiceTag string `json:"ServiceTag"`
+}
+
+type stagOem struct {
+	DELL stagDellData `json:"Dell"`
+}
+
+type dellData struct {
+	ContextID string `json:"ContextID"`
+	Label     string `json:"Label"`
+	Source    string `json:"Source"`
+	FQDD      string `json:"FQDD"`
+}
+
+type oemData struct {
+	DELL dellData `json:"Dell"`
+}
+
+type MetricValue struct {
+	MetricID  string  `json:"MetricId"`
+	TimeStamp string  `json:"Timestamp"`
+	Value     string  `json:"MetricValue"`
+	OEM       oemData `json:"Oem"`
+}
+
+type Resp struct {
+	MType          string `json:"@odata.type"`
+	MContext       string `json:"@odata.context"`
+	MId            string `json:"@odata.id"`
+	Id             string `json:"Id"`
+	Name           string `json:"Name"`
+	ReportSequence string `json:"ReportSequence"`
+	Timestamp      string `json:"Timestamp"`
+	MetricValues   []MetricValue
+	Count          int     `json:"MetricValues@odata.count"`
+	ServiceTag     stagOem `json:"Oem"`
+}
+
+func checkHeader(tmp []byte) bool {
+
+	idPresent := bytes.Contains(tmp, []byte("id: "))
+	dataPresent := bytes.Contains(tmp, []byte("data: "))
+	namePresent := bytes.Contains(tmp, []byte("Name"))
+
+	if idPresent == true && dataPresent == true && namePresent == true {
+		return true
+	} else {
+		return false
+	}
+}
+func InsertMetricReportData(es *elasticsearch.Client, index1 int, tmp []byte) {
+	var res Resp
+	mtx.Lock()
+	reportCount++
+	err := json.Unmarshal(tmp, &dat)
+	mtx.Unlock()
+	// NOTE: For the time being I am using the json unmarshall function as a validator for the parsed metric reports
+	if err != nil {
+		// Don't panic, but keep a count of these invalid packets for analysis
+		// panic(err)
+		invalidJsonPkts++
+		return
+	}
+	err = json.Unmarshal(tmp, &res)
+	if err != nil {
+		panic(err)
+	}
+	fmt.Printf("The size of the received buffer in bytes: %d, The number of Metric Reports processed: %d\n", len(tmp), reportCount)
+	esIndex := res.Id
+	fmt.Printf("Metric Report Name: %s\n", esIndex)
+	for _, value := range res.MetricValues {
+		docIndex := value.MetricID
+		//fmt.Printf("MetricId: %s\n", docIndex)
+		fmt.Println("\nDOC _id:", docIndex)
+		//fmt.Println(value)
+		docval, _ := json.Marshal(value)
+		request := esapi.IndexRequest{Index: strings.ToLower(esIndex), Body: strings.NewReader(string(docval)), DocumentID: docIndex, Refresh: "true"}
+		res, err := request.Do(context.Background(), es)

```

Comment:
It looks like the same request is made with each Metric Value in a Report.    Is that needed?